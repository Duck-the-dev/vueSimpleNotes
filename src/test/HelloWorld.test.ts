import HelloWorldVue from '../components/HelloWorld.vue'
import { mount } from '@vue/test-utils'

describe('counter', () => {
  test('mount component', async () => {
    expect(HelloWorldVue).toBeTruthy()

    const wrapper = mount(HelloWorldVue, {
      props: {
        msg: 'LonesomeT0wn start',
      },
    })

    expect(wrapper.text()).toContain('0')
    expect(wrapper.html()).toMatchSnapshot()

    await wrapper.get('button').trigger('click')

    expect(wrapper.text()).toContain('1')

    await wrapper.get('button').trigger('click')

    expect(wrapper.text()).toContain('2')
  })
})
