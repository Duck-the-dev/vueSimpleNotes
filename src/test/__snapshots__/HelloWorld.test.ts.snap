// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`counter > mount component 1`] = `
"<h1 data-v-e17ea971=\\"\\" class=\\"bg-transparent bg-gradient-to-r from-pink-900 to-purple-500 bg-clip-text text-transparent shadow-purple-400 drop-shadow-md duration-200 hover:shadow-xl hover:drop-shadow-2xl\\">LonesomeT0wn start</h1>
<div data-v-e17ea971=\\"\\" class=\\"card\\"><button data-v-e17ea971=\\"\\" class=\\"btn-primary btn\\" type=\\"button\\"> count is 0</button>
  <p data-v-e17ea971=\\"\\"> Edit <code data-v-e17ea971=\\"\\">components/HelloWorld.vue</code> to test HMR </p>
</div>
<p data-v-e17ea971=\\"\\"> Check out <a data-v-e17ea971=\\"\\" href=\\"https://vuejs.org/guide/quick-start.html#local\\" target=\\"_blank\\">create-vue</a>, the official Vue + Vite starter </p>
<p data-v-e17ea971=\\"\\"> Install <a data-v-e17ea971=\\"\\" href=\\"https://github.com/johnsoncodehk/volar\\" target=\\"_blank\\">Volar</a> in your IDE for a better DX </p>
<p data-v-e17ea971=\\"\\" class=\\"read-the-docs\\">Click on the Vite and Vue logos to learn more</p>"
`;
